crates/storage/src/connection.rs

```diff
@@ -504,11 +540,19 @@ impl<'inner> Transaction<'inner> {
         signature::signature(self, block)
     }
 
+    pub fn increment_reorg_counter(&self) -> anyhow::Result<()> {
+        reorg_counter::increment_reorg_counter(self)
+    }
+
+    pub fn reorg_counter(&self) -> anyhow::Result<ReorgCounter> {
+        reorg_counter::reorg_counter(self)
+    }
+
```

Comment:
Wait, that's actually a bug here: I seem to have lost the code incrementing the reorg counter during a reorg... :(

Will check if we can have a better connection API instead of `purge_block` that would hide details in storage.

Reply:
We definitely can improve matters, but I would leave that for a separate PR. 

I imagine the API would be:
```rust
/// Removes all block data including the tail.
Transaction::reorg(&self, reorg_tail: BlockNumber) -> anyhow::Result<()>;
```

Reply:
I think we can put abstracting this better on the todo pile near the back :)